lib/grinder_files.dart

```diff
@@ -98,12 +100,210 @@ class FileSet {
+   * not yet exist.
+   */
+  Link get asLink => new Link(path);
+
+  /**
+   * Copy the the entity to the given destination. Return the newly created
+   * [Path].
+   */
+  Path copy(Path destDir) {
+    _copyImpl(entity, destDir.asDirectory);
+    return new Path(destDir).join(name);
+  }
+
+  /**
+   * Delete the entity at the path.
+   */
+  void delete() => _deleteImpl(entity);
+
+  /**
+   * Synchronously create the file. See also [File.create].
+   *
+   * If [recursive] is false, the default, the file is created
+   * only if all directories in the path exist. If [recursive] is true, all
+   * non-existing path components are created.
+   */
+  void createFile({bool recursive: false}) {
+    asFile.createSync(recursive: recursive);
+  }
+
+  /**
+   * Synchronously create the directory. See also [Directory.create].
+   *
+   * If [recursive] is false, the default, the file is created
+   * only if all directories in the path exist. If [recursive] is true, all
+   * non-existing path components are created.
+   */
+  void createDirectory({bool recursive: false}) {
+    asDirectory.createSync(recursive: recursive);
+  }
+
+  /**
+   * Synchronously create the link. See also [Link.create].
+   *
+   * If [recursive] is false, the default, the file is created
+   * only if all directories in the path exist. If [recursive] is true, all
+   * non-existing path components are created.
+   */
+  void createLink(String target, {bool recursive: false}) {
+    asLink.createSync(target, recursive: recursive);
+  }
+
+  /**
+   * Join the given path elements to this path, and return a new [Path] object.
+   */
+  Path join([arg0, String arg1, String arg2, String arg3, String arg4,
+    String arg5, String arg6, String arg7, String arg8, String arg9]) {
+    List args = [];
+
+    if (arg0 is List) {
+      args = arg0;
+    } else if (arg0 is String) {
+      _addNonNull(args, arg0);
+      _addNonNull(args, arg1);
+      _addNonNull(args, arg2);
+      _addNonNull(args, arg3);
+      _addNonNull(args, arg4);
+      _addNonNull(args, arg5);
+      _addNonNull(args, arg6);
+      _addNonNull(args, arg7);
+      _addNonNull(args, arg8);
+      _addNonNull(args, arg9);
+    }
+
+    if (args.isEmpty) {
+      return this;
+    } else {
+      return new Path('${path}${_sep}${args.join(_sep)}');
+    }
+  }
+
+  bool operator ==(other) {
+    return other is Path ? path == other.path : false;
+  }
+
+  int get hashCode => path.hashCode;
+
+  String toString() => path;
+
+  static String _coerce(arg) {
+    if (arg is String) {
+      if (arg.length > 1 && arg.endsWith((_sep))) {
+        return arg.substring(0, arg.length - 1);
+      } else {
+        return arg;
+      }
+    }
+    if (arg is FileSystemEntity) return arg.path;
+    if (arg is Path) return arg.path;
+    throw new ArgumentError('expected a FileSystemEntity or a String');
+  }
+}
+
+/**
  * Return the last segment of the file path.
  */
 String fileName(FileSystemEntity entity) {
```

Comment:
This method has been around for a while, and hasn't changed for this PR. I'd have been _much_ happier if FileSystemEntity has just had a `name` property...